src/wallet/db.cpp

```diff
@@ -81,29 +101,24 @@ void CDBEnv::Reset()
     fMockDb = false;
 }
 
-CDBEnv::CDBEnv()
+CDBEnv::CDBEnv(const fs::path& dir_path) : strPath(dir_path.string())
 {
     Reset();
 }
 
 CDBEnv::~CDBEnv()
 {
-    EnvShutdown();
+    Close();
 }
 
-void CDBEnv::Close()
-{
-    EnvShutdown();
-}
-
-bool CDBEnv::Open(const fs::path& pathIn)
+bool CDBEnv::Open()
 {
     if (fDbEnvInit)
         return true;
 
     boost::this_thread::interruption_point();
 
-    strPath = pathIn.string();
+    fs::path pathIn = strPath;
```

Comment:
Lines 32:33 in db.h states
`    // Don't change into fs::path, as that can result in
    // shutdown problems/crashes caused by a static initialized internal pointer.`
When originally written, this comment referred to boost::filesystem. I'm not sure if this comment is still relevant but I did want to bring attention to it.

Reply:
> When originally written, this comment referred to boost::filesystem.

The comment is actually still referring to boost::filesystem, `fs` is just an alias: https://github.com/bitcoin/bitcoin/blob/62fdf9b07087b80d2142799bdd2324f61483359d/src/fs.h#L16

But it's fine to create a path object here. The comment is just explaining why the path objects aren't stored permanently (they stop being usable during shutdown).